tools/testing/test_selections.py

```diff
@@ -37,6 +42,7 @@
     except subprocess.CalledProcessError as e:
         # The safe default for ROCm GHA runners is to run tests serially.
         NUM_PROCS = 1
+    NUM_PROCS_FOR_SHARDING_CALC = 1
```

Comment:
If I read this correctly, `NUM_PROCS_FOR_SHARDING_CALC` will always be set to 1 for ROCm with or without IS_MEM_LEAK_CHECK.  Are there any bad side effects when doing that?

Reply:
@clee2000 ROCm CI runners have a minimum of 2 GPUs, and we intend to keep it that way. Keeping that in mind, wouldn't it be better to set `NUM_PROCS_FOR_SHARDING_CALC` to 2?

cc @jeffdaily